### ISSUE TITLE
Code

### ISSUE THREAD

**Issue by vietpho** [ShihaoShao-GH/SuperGlobal#8]
Hello again!  
Thanks for your helpful response to my previous question. Before building my own pipeline, I'm studying your code, but some parts are unclear.

1. Regional-GeM
In the `rgem.py` file, the code looks like this:

```
class rgem(nn.Module):
    """ Reranking with maximum descriptors aggregation """
    def __init__(self, pr=2.5, size=5):
        super(rgem, self).__init__()
        self.pr = pr
        self.size = size
        self.lppool = nn.LPPool2d(self.pr, int(self.size), stride=1)
        self.pad = nn.ReflectionPad2d(int((self.size-1)//2))
    def forward(self, x):
        nominator = (self.size ** 2)
```

I thought `nominator` was for normalization, but I don't understand the role of `**(1./self.pr)`.

2. Paper and Code Alignment
In your `RerankwMDA.py` file, both `X1` and `X2` appear to be refined. According to the paper, I thought `res_rerank` would correspond to S2 and be computed with the original global descriptor(g_d), not the refined one(g_dr).

I'm sure I'm missing some details, so I'd appreciate your explanation of these aspects.

Thank you!

**Comment by ShihaoShao-GH** [comment]
Hello, I just came back from Paris for ICCV. 

Regards your question 1:

The formulation of `torch.lppool` is $(\sum x^p)^{\frac{1}{p}}$. And what we want is $(\sum\frac{x^p}{wh})^{\frac{1}{p}}$. If we let $wh$ be $(wh)^{\frac{1}{p}}$, the equation will go for $(\sum(\frac{x}{(wh)^{\frac{1}{p}}})^p))^{\frac{1}{p}} = (\sum(\frac{x^p}{wh}))^{\frac{1}{p}}$, and that is what we need.



Your question 2:

Yes, you are right. Sorry we did not make the paper clearly describing this detail.
